homeassistant/components/trend/binary_sensor.py

```diff
@@ -178,7 +180,7 @@ def trend_sensor_state_listener(event):
                     state = new_state.attributes.get(self._attribute)
                 else:
                     state = new_state.state
-                if state not in (STATE_UNKNOWN, STATE_UNAVAILABLE):
+                if state not in (STATE_UNKNOWN, STATE_UNAVAILABLE, STATE_NOT_SET):
```

Comment:
We should fix the proximity sensor instead to return `None` in this case which will be converted to unknown state.

Reply:
OK. I asked this question on the core-dev discord yesterday and this was the proposed solution. Let me close this PR and I'll check the proximity sensor again.